### PR TITLE
Support arm64 native CppUnitTestFramework with `dotnet test`

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Support `dotnet test nativeArm64.dll`
image

### CODE REVIEW

**Review by Evangelink — APPROVED** [review]

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs [hunk 1]
```diff
@@ -408,7 +408,7 @@ public Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, Ca
 
             // We silently force x64 only if the target architecture is the default one and is not specified by user
             // through --arch or runsettings or -- RunConfiguration.TargetPlatform=arch
-            bool forceToX64 = SilentlyForceToX64() && _runsettingHelper.IsDefaultTargetArchitecture;
+            bool forceToX64 = SilentlyForceToX64(sourcePath) && _runsettingHelper.IsDefaultTargetArchitecture;
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs:411]: Is this in this order on purpose? SilentlyForceToX64 should be evaluated even if IsDefaultTargetArchitecture is true?

**Review by nohwnd — APPROVED** [review]

**Comment by nohwnd** [comment]
@MarcoRossignoli  not sure how to revoke my approval without blocking you from merging. So I just disabled automerge. 

Should the native detection run when we know that the default architecture is set?

**Comment by MarcoRossignoli** [comment]
> Should the native detection run when we know that the default architecture is set?

we should never force if native.

File: src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs:411]: > Is this in this order on purpose? SilentlyForceToX64 should be evaluated even if IsDefaultTargetArchitecture is true?

do you meant `false`

**Pull request merged**